chapter04/chapter04.adoc

```diff
@@ -0,0 +1,317 @@
+        name:
+          type: string
+        description:
+          type: string
+        price:
+          format: double
+          type: number
+----
+
+As we can see, MicroProfile OpenAPI has generated API documentation for our resource class. We can use this documentation to learn about the API and how to use it. 
+
+MicroProfile OpenAPI allows developers to produce these specifications directly from their codebase, leveraging annotations and/or providing OpenAPI documents statically. This direct generation ensures that the API documentation is always up to date with the code.
+
+=== Exploring the APIs using Swagger UI
+
+To open Swagger UI for the API documentation generated using MicroProfile OpenAPI, you will need to deploy your application to a server that supports MicroProfile, such as Open Liberty, WildFly, Quarkus, or Payara Micro. These servers automatically generate the OpenAPI documentation for your RESTful services based on the annotations in your code. 
+
+Next, visit the following URL to launch the Swagger UI: 
+----
+http://localhost:8080/openapi/ui
+----
+
+Swagger UI is then used to render this documentation in a user-friendly web interface. Below is the screenshot of swagger UI for the Product REST Resource. 
+
+:figure-caption: Swagger UI 
+.Swagger UI
+image::../images/figure4-1.png[Swagger UI]
+
+== Annotations 
+
+The MicroProfile OpenAPI annotations can be used to document any Jakarta Restful Web Services resource. The annotations can also be used in conjunction with other Jakarta Restful Webservices annotations, such as @Path and @Produces. The most common annotations that are used to document RESTful web services are list in Table 4-1.
+
+[cols="1,3", options="header"]
+|===
+| Annotations | Details
+
+| @OpenAPIDefinition
+| Provides metadata about the entire API. It can include information such as the title, description, version, terms of service, and contact information.
+
+| @Info 
+| Used inside @OpenAPIDefinition to provide API metadata like title, version, description.
+
+| @Contact
+| Specifies contact information for the API, used within @Info.
+
+| @License 
+| Defines the license information for the API, also used within @Info.
+
+| @Operation
+| Describes a single API operation on a resource.
+
+| @APIResponse
+| It is used to document a response from an operation.
+
+| @APIResponses
+| A container for multiple @APIResponse annotations, allowing documentation of different responses for a single API operation.
+
+| @RequestBody
+| Describes the request body of an HTTP request, specifying the content of the body and whether it is required.
+
+| @Schema
+| Provides schema details for a response or request body, specifying the data type, format, and constraints.
+
+| @Parameter
+| Provides information on parameters to the operation, including query parameters, header parameters, and path parameters.
+
+| @ParameterIn
+| Specifies the location of a request parameter, such as `QUERY`, `PATH`, etc.
```

Comment:
`ParameterIn` is an enum used within `@Parameter` rather an annotation itself.

Reply:
Fixed with https://github.com/ttelang/microprofile-tutorial/commit/81a5cb5c93271adad591991f8e8af694e387edb7